src/shammath/include/shammath/matrix_exponential.hpp

```diff
@@ -210,11 +212,25 @@ namespace shammath {
         return coefs;
     }
 
+    /**
+     * @brief this function compute the Taylor's polynomial order (m_star)
+     * the optimal number of matrix product during the taylor evaluation step(k_star)
+     * and the optimal scaling factor (s_star)
+     * @param K maximum number of matrix product allow
+     * @param seq_mk precompute set of Polynomial order
+     * @param seq_theta_mk precomopute set of precompute parameters
+     * @param A the matrix
+     * @param size_A the matrix A size
+     * @param k_start the optimal number of matrix product during the taylor evaluation step
+     * @param m_start the Taylor's polynomial order
+     * @param s_star the optimal scaling factor
+     */
+    template<class T, class Extents1, class Layout1, class Accessor1>
     inline void order_scale(
-        const int K,
+        const i64 K,
```

Comment:
```suggestion
        const i32 K,
```
double precision int can have a strong impact in some cases on GPU. Especially it takes double the amount of registers. Knowing that the matrix exponential may be subject to a lot of register pressure it is maybe best to keep single precision indexing `i32`.